packages/@aws-cdk/aws-appmesh/lib/gateway-route.ts

```diff
@@ -0,0 +1,354 @@
+import * as cdk from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnGatewayRoute } from './appmesh.generated';
+import { Protocol } from './shared-interfaces';
+import { IVirtualGateway } from './virtual-gateway';
+import { IVirtualService } from './virtual-service';
+
+/**
+ * Interface for which all Gateway Route based classes MUST implement
+ */
+export interface IGatewayRoute extends cdk.IResource {
+  /**
+   * The name of the Gateway Route
+   *
+   * @attribute
+   */
+  readonly gatewayRouteName: string,
+
+  /**
+   * The Amazon Resource Name (ARN) for the Gateway Route
+   *
+   * @attribute
+   */
+  readonly gatewayRouteArn: string;
+}
+
+interface GatewayRouteBaseProps {
+  /**
+   * The name of the Gateway Route
+   *
+   * @default - an automatically generated name
+   */
+  readonly gatewayRouteName?: string;
+
+  /**
+   * The VirtualService this Gateway Route directs traffic to
+   */
+  readonly routeTarget: IVirtualService;
+}
+
+/**
+ * Properties to define new Gateway Routes
+ */
+export interface GatewayRouteProps extends GatewayRouteBaseProps {
+  /**
+   * The Virtual Gateway this Gateway Route is associated with
+   */
+  readonly virtualGateway: IVirtualGateway;
+}
+
+/**
+ * Base interface for HTTP Based Gateway Routes
+ */
+export interface GatewayHttpRouteBaseProps extends GatewayRouteBaseProps {
+  /**
+   * The criterion for determining a request match for this Gateway Route.
+   *
+   * @default - prefix match on "/"
+   */
+  readonly match?: GatewayHttpRouteMatch;
+  /**
+   * HTTP Procol
+   */
+  readonly routeType: Protocol.HTTP;
+}
+
+/**
+ * Interface for HTTP Based Gateway Routes
+ */
+export interface GatewayHttpRouteProps extends GatewayHttpRouteBaseProps, GatewayRouteProps {}
+
+/**
+ * Base interface for HTTP2 Based Gateway Routes
+ */
+export interface GatewayHttp2RouteBaseProps extends GatewayRouteBaseProps {
+  /**
+   * The criterion for determining a request match for this Gateway Route.
+   *
+   * @default - prefix match on "/"
+   */
+  readonly match?: GatewayHttp2RouteMatch;
```

Comment:
I believe we can actually use the same `HttpRouteMatch` class here. They use the same underlying cfn resource https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-appmesh-gatewayroute-gatewayroutespec.html